File: app/src/main/java/net/chris/demo/guardian/ui/recycler/ContentListItem.java
package net.chris.demo.guardian.ui.recycler;

public class ContentListItem {

    private String id;

    private String sectionName;

    private String webPublicationDate;

    private String webTitle;

    private String webUrl;

    private boolean isSelected;

    private ContentListItem(String id, String sectionName, String webPublicationDate, String webTitle, String webUrl, boolean isSelected) {
        this.id = id;
        this.sectionName = sectionName;
        this.webPublicationDate = webPublicationDate;
        this.webTitle = webTitle;
        this.webUrl = webUrl;
        this.isSelected = isSelected;
    }

    public String getId() {
        return id;
    }

    public String getSectionName() {
        return sectionName;
    }

    public String getWebPublicationDate() {
        return webPublicationDate;
    }

    public String getWebTitle() {
        return webTitle;
    }

    public String getWebUrl() {
        return webUrl;
    }

    public boolean isSelected() {
        return isSelected;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setSectionName(String sectionName) {
        this.sectionName = sectionName;
    }

    public void setWebPublicationDate(String webPublicationDate) {
        this.webPublicationDate = webPublicationDate;
    }

    public void setWebTitle(String webTitle) {
        this.webTitle = webTitle;
    }

    public void setWebUrl(String webUrl) {
        this.webUrl = webUrl;
    }

    public void setSelected(boolean selected) {
        isSelected = selected;
    }

    public static class Builder {
        private String id;
        private String sectionName;
        private String webPublicationDate;
        private String webTitle;
        private String webUrl;
        private boolean isSelected;

        public Builder setId(String id) {
            this.id = id;
            return this;
        }

        public Builder setSectionName(String sectionName) {
            this.sectionName = sectionName;
            return this;
        }

        public Builder setWebPublicationDate(String webPublicationDate) {
            this.webPublicationDate = webPublicationDate;
            return this;
        }

        public Builder setWebTitle(String webTitle) {
            this.webTitle = webTitle;
            return this;
        }

        public Builder setWebUrl(String webUrl) {
            this.webUrl = webUrl;
            return this;
        }

        public Builder setSelected(boolean selected) {
            this.isSelected = selected;
            return this;
        }

        public ContentListItem build() {
            return new ContentListItem(id, sectionName, webPublicationDate, webTitle, webUrl, isSelected);
        }
    }
}

File: app/src/main/kotlin/net/chris/demo/guardian/ui/recycler/ContentListItem.kt
package net.chris.demo.guardian.ui.recycler

class ContentListItem private constructor(
        var id: String?,
        var sectionName: String?,
        var webPublicationDate: String?,
        var webTitle: String?,
        var webUrl: String?,
        var isSelected: Boolean
) {

    class Builder {
        private var id: String? = null
        private var sectionName: String? = null
        private var webPublicationDate: String? = null
        private var webTitle: String? = null
        private var webUrl: String? = null
        private var isSelected: Boolean = false

        fun setId(id: String): Builder {
            this.id = id
            return this
        }

        fun setSectionName(sectionName: String): Builder {
            this.sectionName = sectionName
            return this
        }

        fun setWebPublicationDate(webPublicationDate: String): Builder {
            this.webPublicationDate = webPublicationDate
            return this
        }

        fun setWebTitle(webTitle: String): Builder {
            this.webTitle = webTitle
            return this
        }

        fun setWebUrl(webUrl: String): Builder {
            this.webUrl = webUrl
            return this
        }

        fun setSelected(selected: Boolean): Builder {
            this.isSelected = selected
            return this
        }

        fun build(): ContentListItem {
            return ContentListItem(id, sectionName, webPublicationDate, webTitle, webUrl, isSelected)
        }
    }
}
